jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlConfiguration.java

```diff
@@ -332,14 +377,22 @@ public void initializeDefaults(Object object)
 
     private static class JettyXmlConfiguration implements ConfigurationProcessor
     {
-        private String _url;
+        private String _location;
         XmlParser.Node _root;
         XmlConfiguration _configuration;
 
+        @Override
+        public void init(Resource resource, XmlParser.Node root, XmlConfiguration configuration)
+        {
+            _location = resource == null ? "UNKNOWN_RESOURCE" : resource.toString();
```

Comment:
I don't like storing as fields default Strings that are used only in log statements.
I prefer the field to be the `Resource` itself, and then add a `toString()` method to `JettyXmlConfiguration` that formats the resource and modify the log statements to log `this` instead of `_location`.
I don't care much about the difference between `UNKNOWN_RESOURCE` and `UNKNOWN_URL` in debug statements - it's unknown anyway.

Reply:
Will do.